### ISSUE TITLE
Reports the line number in case of error

### ISSUE THREAD

**Issue by jcornaz** [jcornaz/beancount-parser#11]
As a end-user, if there is an error in my beancount file, I would like to know at which line the error was found, so I have an easier time finding my mistakes.

---

Currently, the main `Error` type contains no information. Needless to say, that is not very helpful :smile:.

I think it should at least contain the line number. 

---

### Some potentially useful links

* https://github.com/rust-bakery/nom/blob/main/doc/error_management.md
* https://docs.rs/nom_locate/latest/nom_locate

**Comment by TTWNO** [comment]
This could also be useful in higher-level crates as part of the Directive struct. Is there any way you could include the line number(s) of the directive so that it's easier for my crate to have an error like the following:

```rust
TransactionValidityError::UnbalancedTransaction(transaction: Transaction, line_start: i32/* optional: , line_end: i32 */)
```

Of course, this would only be possible if the Directive enum, or every variant, included the line numbers where they were parsing in the first place. Is that something you'd be interested in a PR for?

**Comment by jcornaz** [comment]
Yes, I thought about that too. Even a successful directive parsing should include a line number so that higher-level errors can still refer to the line number.

**Comment by jcornaz** [comment]
As mentioned [here](https://github.com/jcornaz/beancount-parser/discussions/38), I am now working on a rewrite of the parser: https://github.com/jcornaz/beancount_parser_2

And, `beancount_parser_2` already features line numbers on errors and directives.

I don't plan to work more on the present implementation unless to fix reported bugs and review/merge submitted pull requests.

This is why I now close this issue as "not planned". But I'll still accept a pull request implementing this feature if someone needs it.

**Comment by jcornaz** [comment]
Solved in version `2.0.0-beta.1`
